python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+
+    def run(self):
+        """ Not implemented in this subtask. Please use the store and load
+        methods.
+        """
+        Raise(NotImplementedError)
+
+    def commit(self):
+        """ Save changes to the sqlite database.
+        """
+        self._db_connection.commit()
+
+    def close(self):
+        """ Close the connection to the sqlite database.
+        """
+        self._db_connection.close()
+
+    def create_tables(self):
+        """ If no sqlite database exists with the correct tables we can create
+        one using this method.
+
+        Returns
+        -------
+        bool
+            Successfully created a new database with specified tables.
+        """
+
+        obj_schema = make_minimal_dia_object_schema()
+        src_schema = make_minimal_dia_source_schema()
+
+        self._db_cursor.execute(
+            'select name from sqlite_master where type = "table"')
+        db_tables = self._db_cursor.fetchall()
+
+        if db_tables:
+            return False
+        else:
+            name_type_string = ""
+            for sub_schema in obj_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects (%s)" % name_type_string)
+
+            name_type_string = ""
+            for sub_schema in src_schema:
+                tmp_name = sub_schema.getField().getName()
+                tmp_type = afw_to_db_types[
+                    sub_schema.getField().getTypeString()]
+                if tmp_name == 'id':
+                    tmp_type += " PRIMARY KEY"
+                name_type_string += "%s %s," % (tmp_name, tmp_type)
+            name_type_string = name_type_string[:-1]
+            self._db_cursor.execute(
+                "CREATE TABLE dia_sources (%s)" % name_type_string)
+
+            self._db_cursor.execute(
+                "CREATE TABLE dia_objects_to_dia_soruces "
+                "(src_id INTEGER PRIMARY KEY, obj_id INTEGER)")
+
+            self._db_connection.commit()
+
+        return True
+
+    @pipeBase.timeMethod
+    def load(self, ctr_coord, radius):
+        """ Load all DIAObjects and associated DIASources.
+
+        Parameters
+        ----------
+        ctr_coord : lsst.afw.geom.Coord
+            Center position of on on sky circle to load.
+        radius : lsst.afw.geom.Angle
+
+
+        Returns
+        -------
+        lsst.ap.association.DIAObjectCollection
+        """
+        indexer_indices, on_boundry = self.indexer.get_pixel_ids(
+            ctr_coord, radius)
+
+        dia_objects = self.get_dia_objects(indexer_indices)
+
```

Comment:
I'm allowing the load here to be a bit greedy and letting the code keep all of the objects in the pixels that cover the circle. This shouldn't be too bad as default resolutions are small enough that I don't think they represent too much extra area compared to input circle. The worse thing I'm doing currently is not cutting the pixels to the current ccd being processed. Once we get AssociationTask running in ap_verify, we'll definitely want to change this.

Reply:
This is (I assume) a public method. If it gets called by something that assumes the method works exactly as advertised (e.g., because it needs all sources to be within some radius) then you'll have trouble. But if you're planning to change it soon then we can worry about it then.

Reply:
I'll add a comment that the return is approximate.